sidechain/state/src/impls.rs

```diff
@@ -0,0 +1,162 @@
+//! Implement the sidechain state traits. This is only sgx-compatible due to [`SgxExternalities`]
+//! missing some features in `std`.
+
+use crate::{sgx_reexports::*, Error, SidechainDB, SidechainState, StateHash, StateUpdate};
+use codec::{Decode, Encode};
+use itp_storage::keys::storage_value_key;
+use log::error;
+use sgx_externalities::SgxExternalitiesTrait;
+use sp_core::{hashing::blake2_256, H256};
+use sp_std::prelude::Vec;
```

Comment:
is this import done so intentionally, rather than using `std::vec::Vec`?

Reply:
Not really. It's just convenient if the crate would not use `sgx_tstd` in `no_std` otherwise.